### DISCUSSION TITLE
Debian packaging situation

### DISCUSSION THREAD

**Discussion by Grounded0** [dosbox-staging/dosbox-staging#3806] [General]
If anyone wants to take a stab at DOSBox Staging Debian package my bounty ticket has links to Debian ITP and ITS for DOSBox Staging.

The payout for some mailing list action could be pretty good. You can use pre-existing DOSBox Staging Ubuntu PPA package as a template for creating DOSBox Staging Debian package so much of the technical bits are already figured out for you. Link is on the ticket.

DOSBox Staging man page which we rewrote is now Debian quality and there really isn't any constraints on the technical side of things or those can be fairly easily worked out. What is needed is persistence to work with Debian bureaucracy to get things moving and I think 500 USD is adequate compensation for the effort.

Connected issues and PRs:

- #3677

**Comment by hosiet** [comment]
Do you have answers to https://bugs.debian.org/973822#60 readily available?

**Comment by interloper98** [comment]
I'll be PR'ing:
 - `debian/*` records that the Debian tools need to create a package
 - These records will be dynamically constructed from existing metadata in the project to avoid duplication and maintenance burden.
 - `scripts/build_deb_package.sh`, to generate an .orig tarball and build the .deb package
 - workflow additions (probably using the existing Linux Docker workflow) to create a .deb artifact

Shared/existing Debian libraries will be used to the maximum extent they're available.

The priority is on the user: no features will be stripped: Debian users will be able to install it with confidence that they have all the working features that other DOSBox Staging users get - and therefore will be able to play game packs without things breaking or failing.

The user will be able to install the .deb like this:

`sudo apt install ./dosbox-staging_0.82.0~alpha+gitd153d2234-1_amd64.deb`

Apt will then install all dependencies plus DOSBox Staging to the host.

Here's an info dump:

**dpkg --info ../dosbox-staging_0.82.0~alpha+gitd153d2234-1_amd64.deb**

```
 new Debian package, version 2.0.
 size 2891282 bytes: control archive=9161 bytes.
     647 bytes,    12 lines      control
   31389 bytes,   365 lines      md5sums
 Package: dosbox-staging
 Version: 0.82.0~alpha+gitd153d2234-1
 Architecture: amd64
 Maintainer: Your Name <your.email@example.com>
 Installed-Size: 16675
 Depends: libasound2t64 (>= 1.0.16), libc6 (>= 2.38), libfluidsynth3 (>= 2.2.0), libgcc-s1 (>= 4.0), libgl1, libopusfile0 (>= 0.5), libpng16-16t64 (>= 1.6.2), libsdl2-2.0-0 (>= 2.26.0), libsdl2-net-2.0-0 (>= 2.0.1), libslirp0 (>= 4.1.0), libspeexdsp1 (>= 1.2.1), libstdc++6 (>= 13), zlib1g (>= 1:1.1.4)
 Section: devel
 Priority: optional
 Homepage: TBD
 Description: Short description (TBD)
  Long description: TBD
```

- I don't care if this is adopted by Debian's repo maintainers.
- I won't be asking for payment from @Grounded0.

**Reply by johnnovak** [reply]
Thanks for doing this @interloper98 ❤️

**Reply by Torinde** [reply]
> You can use pre-existing DOSBox Staging Ubuntu PPA package as a template

What I find at [Staging webiste](https://www.dosbox-staging.org/releases/linux/#building-from-source):
> Ubuntu and Mint repository package[¶](https://www.dosbox-staging.org/releases/linux/#ubuntu-and-mint-repository-package)
Available via [Personal Package Archive](https://launchpad.net/~feignint/+archive/ubuntu/dosbox-staging)

leads to a PPA page stopping at 0.78.1

I don't know what's the relationship between Debian/Ubuntu repositories, but just mentioning in case it's related and you're interested in fixing both together.

**Comment by Grounded0** [comment]
@hosiet 

I think most of the questions are addressed here:

https://github.com/dosbox-staging/dosbox-staging/issues/3677#issuecomment-2142105816

If those answers are deemed inadequate DOSBox Staging Debian package can also strip out those binary blobs. DOSBox Staging binary blobs are sourced from the FreeDOS Project.

DOSBox-X Debian package also ships with binary blobs but they're hidden in the source code so I don't see a problem here (`DEVICE.COM` for example):

https://github.com/joncampbell123/dosbox-x/blob/master/src/builtin/device_com.cpp

**Comment by interloper98** [comment]
PR incoming! https://github.com/dosbox-staging/dosbox-staging/pull/3809

**Comment by Grounded0** [comment]
@interloper98 

Nice work.

The ball is definitely off the DOSBox Staging court now.

**Comment by Grounded0** [comment]
For those unaware, this is the process for getting a package into Debian:
- https://github.com/dosbox-staging/dosbox-staging/issues/3677#issuecomment-2241282483

**Comment by Grounded0** [comment]
Follow-up discussions to the bounty ticket:

- #3677

**Comment by Torinde** [comment]
@nerun, FYI

**Reply by johnnovak** [reply]
> @nerun, FYI

Not sure why you're tagging the person? The bounty has been revoked, and Grounded0 has semi-retired from the project.

**Reply by Torinde** [reply]
[It seems](https://github.com/dosbox-staging/archived-discussions-for-dosbox-staging/discussions/7) he was interested in creating/maintaining a Debian package (even without a bounty).

**Reply by johnnovak** [reply]
> [It seems](https://github.com/dosbox-staging/archived-discussions-for-dosbox-staging/discussions/7) he was interested in creating/maintaining a Debian package (even without a bounty).

Well, that was many years ago...

**Reply by Grounded0** [reply]
All my money is tied up in buying a town house in 2026 with help from my parents. After it has been bought, I'm buying a Saab convertible I've always wanted for the summers.
